### ISSUE TITLE
Python Components start method issues with getSceneList() & createConstraint()

### ISSUE THREAD

**Issue by Maujoe** [UPBGE/blender#335]
I notice that the start method of the python components has some issus in upbge.

When I call bge.logic.getSceneList() in the start method it return an empty list.
The same happends with bge.constraints.createConstraint(..) that return None.
In the update method everything works right.
Here is a file:
[components_start_bug.zip](https://github.com/UPBGE/blender/files/643966/components_start_bug.zip)

When I use agoose77's Addon ([https://blenderartists.org/forum/showthread.php?393763-Python-Components-Addon](url)) I have no issues with the start method. 

I know nothing about the bge code but I think it could be a timing problem in the Initialization state or something like that.

**Comment by sdfgeoff** [comment]
This is because the objects present in a scene are created before the scene is first added.
The simple solution is to do:

```
import bge
from collections import OrderedDict

class loadsequence(bge.types.KX_PythonComponent):
    '''The Player'''
    args = OrderedDict([
    ])
    def start(self, args):
        print("This runs before the scene is fully added to the game")
        self.update = self.init

    def init(self):
        print("This runs on the first frame with everything in the scene")
        self.update = self.run
        print(bge.logic.getSceneList())

    def run(self):
        print("This will run every frame after")
```

**Comment by youle31** [comment]
Precision: I marked it as a bug because it makes my blender crash at ge start.

**Comment by panzergame** [comment]
@Maujoe, @sdfgeoff, @youle31 : Could you confirm the issue is gone and close the issue in consideration ?

**Comment by youle31** [comment]
@panzergame : The crash is fixed for me.

I have this error in the console in Maujoe test file:

Blender Game Engine Started
Debug: Game Controller (XInput Controller) with index 0 initialized
Traceback (most recent call last):
  File "C:\Users\yul\Desktop\components_start_bug.blend\comp.py", line 15, in start
  File "C:\Users\yul\Desktop\components_start_bug.blend\comp.py", line 30, in constraint_test
SystemError: Representation not overridden by object.
Blender Game Engine Finished
Debug: Game Controller (XInput Controller) with index 0 closed

but I don't know if this is a bug or only due to a wrong usage of start method.

**Comment by panzergame** [comment]
@youle31 : Iirc it's because `constraint`variable in the code isn't printable, but it's to check.

**Comment by youle31** [comment]
All right :) This sounds good. I close the issue. Thanks panzergame. Fixed in  and in  eae0eb284a4ca9d757384857468e51e for crash 86adc3c3c53c01885da01716b849479712d37202 for constraints representation
